### PR TITLE
Add keyboard navigation to gallery

### PR DESCRIPTION
**Pull request by heymynameisrob**
**Use keyboard triggers (⬅️ ➡️) to move between images in the gallery.**

When I looked at the photos, my first reaction was to navigate using the keyboard. I could use `Escape` to close the dialog but not keyboard events. Thought it'd be cool to add.

Left a note about preloading the next image so you don't get a flash but left that for now.

### CODE REVIEW

File: src/components/Gallery.astro [hunk 1]
```diff
@@ -211,4 +212,38 @@ const { images } = Astro.props;
...
+    switch (e.key) {
+      case "ArrowRight":
+        e.preventDefault();
+        showImage(e.ctrlKey ? dialogs.length - 1 : currentIndex + 1);
+        break;
+      case "ArrowLeft":
+        e.preventDefault();
+        showImage(e.ctrlKey ? 0 : currentIndex - 1);
+        break;
+      case "Escape":
```
Comment (danjordan) [inline src/components/Gallery.astro:245]: Is this needed?

Comment (heymynameisrob) [reply src/components/Gallery.astro:246]: nope - just me being untidy

File: src/components/Gallery.astro [hunk 2]
```diff
@@ -197,6 +197,7 @@ const { images } = Astro.props;
 <script>
   const buttonsOpen = document.querySelectorAll(".gallery__button");
   const buttonsClose = document.querySelectorAll(".gallery__close");
+  const dialogs = document.querySelectorAll(".gallery__dialog");
```
Comment (danjordan) [inline src/components/Gallery.astro:200]: `const dialogs = document.querySelectorAll<HTMLDialogElement>(".gallery__dialog");` lets you delete all your `as HTMLDialogElement` below.

**Comment by pawelgrzybek** [comment]
thanks for input @danjordan  🫶

**Pull request merged**